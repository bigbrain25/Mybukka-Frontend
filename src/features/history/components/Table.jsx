import React from 'react';
import PropTypes from 'prop-types';
import shortId from 'shortid';
import TableHead from './TableHead';
import TableBody from './TableBody';

const Table = ({ data, handleClick }) => (
  <div
    className="d-flex flex-column flex-xl-column flex-lg-column flex-md-column
    justify-content-between"
  >
    <TableHead />
    {data.map(dom => (
      <TableBody
        key={shortId.generate() + dom.title}
        handleClick={() => handleClick(dom.status)}
        time={dom.time}
        status={dom.status}
        quantity={dom.quantity}
        title={dom.title}
        price={dom.price}
        orderId={dom.orderId}
        courierName={dom.courier.name}
        courierImg={dom.courier.img}
        pickupAddress={dom.pickup.address}
        pickupContactMobile={dom.pickup.contactMobile}
        deliveryAddress={dom.delivery.address}
        deliveryName={dom.delivery.name}
        deliveryContactMobile={dom.delivery.contactMobile}
      />
    ))}
  </div>
);

export default Table;

Table.propTypes = {
  handleClick: PropTypes.func.isRequired,
  data: PropTypes.arrayOf(PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.objectOf(PropTypes.string)
    ])
  )).isRequired
};
