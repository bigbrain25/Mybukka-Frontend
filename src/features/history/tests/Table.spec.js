import React from 'react';
import Table from '../components/Table';

describe.skip('Table component', () => {
  const props = {
    data: [
      {
        time: '27/3/2019 12:43 pm',
        _id: '03d77',
        status: 'pending',
        cart: { items: [{ meals: { title: 'jollof rice', price: 2000 } }] },
        deliveryAddress: {
          address: 'delivery address street',
          name: 'mr delivery name',
          contactMobile: '94574783930',
          time: '27/3/2019 12:43 pm',
        },
        pickup: {
          address: 'pickup address street',
          contactMobile: '03847858',
        },
      }],
    handleClick: jest.fn(),
    extractPrice: () => 300,
    extractQuantity: () => 3,
  };

  const { container, getByTestId } = render(<Table {...props} />);

  it('renders properly', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders properly', () => {
    const tableBtn = getByTestId('table-btn');
    fireEvent.click(tableBtn);
    expect(props.handleClick).toHaveBeenCalled();
  });
});
