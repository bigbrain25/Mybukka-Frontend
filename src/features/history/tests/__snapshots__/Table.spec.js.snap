// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders properly 1`] = `
<div>
  <div
    class="d-flex flex-column flex-xl-column flex-lg-column flex-md-column
 justify-content-between"
  >
    <div
      class="row flex custom-border d-xl-flex d-lg-flex d-md-none d-none"
    >
      <div
        class="col-2 pt-1 pb-1 custom-border bold"
      >
        Status
      </div>
      <div
        class="col-2 pt-1 pb-1 custom-border bold"
      >
        PickUp
      </div>
      <div
        class="col-2 pt-1 pb-1 custom-border bold"
      >
        DropOff
      </div>
      <div
        class="col-2 pt-1 pb-1 custom-border bold"
      >
        Order
      </div>
      <div
        class="col-2 pt-1 pb-1 custom-border bold"
      >
        Price
      </div>
      <div
        class="col-2 pt-1 pb-1 custom-border bold"
      >
        Courier
      </div>
    </div>
    <div
      aria-pressed="false"
      class="row flex flex-xl-row flex-lg-row d-xl-flex d-lg-flex d-md-none
 d-none cursor-pointer custom-border-px"
      data-testid="table-btn"
      role="button"
      tabindex="0"
    >
      <div
        class="col-xl-2 col-lg-2 col-md-10 pt-3 custom-border"
      >
        <div
          class="text-uppercase"
        >
          <h5
            class="font-size-14 text-warning"
          >
            pending
          </h5>
          <h5
            class="light font-size-13 suggested-item-price"
          >
            27/3/2019 12:43 pm
          </h5>
        </div>
      </div>
      <div
        class="col-xl-2 col-lg-2 col-md-10 pt-3 custom-border"
      >
        <div>
          <h5
            class="text-uppercase font-size-14"
          >
            salad and bbq
          </h5>
          <h5
            class="light text-capitalize font-size-13"
          >
            pickup address street
          </h5>
          <h5
            class="light font-size-13"
          >
            03847858
          </h5>
        </div>
      </div>
      <div
        class="col-xl-2 col-lg-2 col-md-10 pt-3 custom-border"
      >
        <div
          class="dropoff"
        >
          <h6
            class="font-size-14 capitalize"
          >
            mr delivery name
          </h6>
          <h6
            class="light font-size-13"
          >
            delivery address street
          </h6>
          <h6
            class="light font-size-13"
          >
            94574783930
          </h6>
        </div>
      </div>
      <div
        class="col-xl-2 col-lg-2 col-md-10 pt-3 custom-border"
      >
        <div
          class="history-order"
        >
          <h5
            class="font-size-14"
          >
            Order 
            03d77
          </h5>
          <h5
            class="light font-size-13"
          >
            2
             Items
          </h5>
        </div>
      </div>
      <div
        class="col-xl-2 col-lg-2 col-md-10 pt-3 custom-border"
      >
        <div
          class="history-price"
        >
          <h5
            class="text-color font-size-14"
          >
            â‚¦
            3000
          </h5>
        </div>
      </div>
      <div
        class="col-xl-2 col-lg-2 col-md-10 pt-3 custom-border"
      >
        <div
          class="courier-history"
        >
          <img
            alt="courier"
            class="courier-img"
            src="https;//www.google.com/img"
          />
          <h6
            class="light"
          >
            curier name
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
